test/controllers/user_role_controller_test.exs

```diff
@@ -60,66 +50,44 @@ defmodule CodeCorps.UserRoleControllerTest do
     test "creates and renders resource when data is valid", %{conn: conn} do
       user = insert(:user)
       role = insert(:role)
-
-      payload = build_payload |> put_relationships(user, role)
-      path = conn |> user_role_path(:create)
-      json = conn |> post(path, payload) |> json_response(201)
-
-      id = json["data"]["id"] |> String.to_integer
-      user_role = UserRole |> Repo.get(id)
-
-      assert json["data"]["id"] == "#{user_role.id}"
-      assert json["data"]["type"] == "user-role"
-      assert json["data"]["relationships"]["user"]["data"]["id"] == "#{user_role.user_id}"
-      assert json["data"]["relationships"]["role"]["data"]["id"] == "#{user_role.role_id}"
+      attrs = (%{user: user, role: role})
+      assert conn |> request_create(attrs) |> json_response(201)
     end
 
     @tag authenticated: :admin
     test "does not create resource and renders errors when data is invalid", %{conn: conn} do
```

Comment:
maybe update the test headings to be more concise, and put '404' in the header of this one